src/mapbox/mapbox.js

```diff
@@ -286,7 +286,15 @@ export default class Mapbox {
   }
 
   _getViewState(props) {
-    return props.viewState || props;
+    const {
+      longitude,
+      latitude,
+      zoom,
+      pitch = 0,
+      bearing = 0,
+      altitude = 1.5
+    } = props.viewState || props;
```

Comment:
Should we drop support for `props` and just look at `props.viewState`?
Or just issue a deprecation warning if `props.viewState` is not set?

Reply:
There is no plan to deprecate flat props in react-map-gl. Doesn't make sense to switch to `viewState` without multi-viewport.